File: app/src/main/java/com/nasahapps/materialdesigntoolbox/example/ui/TestActivity.java
package com.nasahapps.materialdesigntoolbox.example.ui;

import android.os.Bundle;
import android.support.v4.view.ViewPager;

import com.nasahapps.materialdesigntoolbox.components.StepperProgressLayout;
import com.nasahapps.materialdesigntoolbox.example.R;
import com.nasahapps.materialdesigntoolbox.example.ui.adapter.MockViewPagerAdapter;

import butterknife.Bind;

public class TestActivity extends BaseActivity {

    @Bind(R.id.viewPager)
    ViewPager mViewPager;
    @Bind(R.id.layout)
    StepperProgressLayout mLayout;

    @Override
    public int getLayoutId() {
        return R.layout.activity_test;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        mViewPager.setAdapter(new MockViewPagerAdapter(getSupportFragmentManager()));
        mViewPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

            }

            @Override
            public void onPageSelected(int position) {
                mLayout.setProgress(position + 1);
            }

            @Override
            public void onPageScrollStateChanged(int state) {

            }
        });
        mLayout.addOnStepperProgressListener(new StepperProgressLayout.OnStepperProgressListener() {
            @Override
            public void onStepSelected(int position) {
                mViewPager.setCurrentItem(position, true);
            }

            @Override
            public void onStepDeselected(int position) {

            }

            @Override
            public void onStepsFinished() {

            }
        });
    }

}

File: app/src/main/java/com/nasahapps/materialdesigntoolbox/example/ui/TestActivity.kt
package com.nasahapps.materialdesigntoolbox.example.ui

import android.content.Intent
import android.os.Bundle
import com.nasahapps.materialdesigntoolbox.example.R
import com.nasahapps.mdt.bottomsheets.BottomSheetItem
import com.nasahapps.mdt.bottomsheets.BottomSheetUtils
import kotlinx.android.synthetic.main.activity_test.*

class TestActivity : BaseActivity() {

    override fun getLayoutId(): Int {
        return R.layout.activity_test
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        button?.setOnClickListener {
            val launcherIntent = Intent(Intent.ACTION_MAIN)
            launcherIntent.addCategory(Intent.CATEGORY_LAUNCHER)
            val appList = packageManager.queryIntentActivities(launcherIntent, 0)
            val sheetItems = appList.map { BottomSheetItem(it.loadLabel(packageManager), it.loadIcon(packageManager)) }
            BottomSheetUtils.DialogBuilder(this, false)
                    .setTitle("Open in")
                    .setItems(R.menu.menu_bottom_sheet, null)
                    .show()
        }

        shareButton?.setOnClickListener {
            val i = Intent(Intent.ACTION_SEND)
            i.putExtra(Intent.EXTRA_TEXT, "This is my text to send.")
            i.type = "text/plain"
            startActivity(Intent.createChooser(i, "Send to..."))
        }
    }

}
